sway/commands/clipboard.c

```diff
@@ -0,0 +1,38 @@
+#include <wlc/wlc.h>
+#include <unistd.h>
+#include <string.h>
+#include "sway/commands.h"
+#include "stringop.h"
+
+static void send_clipboard(void *data, const char *type, int fd) {
+	if ((strcmp(type, "text/plain") != 0) &&
+			(strcmp(type, "text/plain;charset=utf-8") != 0)) {
+		close(fd);
+		return;
+	}
+
+	const char *str = data;
+	write(fd, str, strlen(str));
+	close(fd);
+}
+
+struct cmd_results *cmd_clipboard(int argc, char **argv) {
+	static const char *current_data = NULL;
+
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "clipboard", EXPECTED_AT_LEAST, 1))) {
+		return error;
+	}
+
+	static const char *types[2] = {
+		"text/plain",
+		"text/plain;charset=utf-8"
+	};
+
+	const char *str = join_args(argv, argc);
+	wlc_set_selection((void*) str, types, 2, &send_clipboard);
+
+	free((void*) current_data);
```

Comment:
Casting to `void*` here is unneccessary and probably on `wlc_set_selection` as well.